src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/legacy_reputation/Ownable.sol';
+import 'ROOT/extensions/MarketExtensions.sol';
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/DisputeBondToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/factories/MapFactory.sol';
+import 'ROOT/factories/ShareTokenFactory.sol';
+import 'ROOT/factories/ReportingTokenFactory.sol';
+import 'ROOT/factories/DisputeBondTokenFactory.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/token/ERC20Basic.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+
+
+contract Market is DelegationTarget, Typed, Initializable, Ownable {
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        PRE_REPORTING,
+        AUTOMATED_REPORTING,
+        AUTOMATED_DISPUTE,
+        LIMITED_REPORTING,
+        LIMITED_DISPUTE,
+        ALL_REPORTING,
+        ALL_DISPUTE,
+        FORK,
```

Comment:
Consider renaming this to `FORKING` to align a bit better with the tense of the others (e.g., `REPORTING`).  Alternatively, consider changing `*_REPORTING` to `*_REPORT`.  The latter would result in more consistent tense for all, but quite as natural.  Your call with however you want to name the enum items, I tend to favor consistency but I don't care enough or have a good naming scheme to push very hard for it.  :)